### DISCUSSION TITLE
Persistent preferences without YAML edits

### DISCUSSION THREAD

**Discussion by damanic** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#86] [Ideas]
@BigBobbas 

Would it be possible to make the UI for S3Box configurable via the esphome webserver service on the device, using persistent variable storage?  This way you could push OTA updates and the user can keep their preferences/settings.  

My issue with the current set up is that if there is an update I would need to edit the yaml and re-apply customisations.

I would like to see a UI for the sbox3 that can be configured through HA/webserver. Sure there would be a compromise in customisation scope but those that want to customise further at yaml level can fork. 

The persistent configuration could cover the button grid where each button can be:
- used as a trigger in HA 
- assigned a icon
- assigned a colour 
- assigned to an entity value (eg. value of temp sensor)

Something as simple as that would cover most peoples needs, where OTA updates improve general integration and system/device level features, and the users persistent config maintains their button grids.

**Comment by BigBobbas** [comment]
Hi, 
Thank you for your feedback and suggestions, 
it is something that I have been looking at over the last few days. 
There is no simple way of acheiving this that I am aware of and have only really been involved with the ESPHome side of things.

There is a possibility of using 'services' now named 'actions' but I need the time to look into this. 

The idea of a grid of  'generic' buttons may also  be an option and automation in HA - possibly a Blueprint for the user to configure these.

This project started as something I made for myself with little knowledge and has grown to become something that people who like to 'tinker' can also do themselves. 

I agree that it would be nice to have this made more easily configurable and I will look further into it. It is just a matter of having the time and inclination to re-write to achieve this. It would be great to use OTA updates to only update any core changes.

LVGL display is now possible on the S3 box so I would possibly look at ways to implement an easier user experience whilst re-writing to use LVGL rather than the standard display component. 

I have almost finished what will be the final standard display version and will be merging this to main this week. 

Then I will look at making a V2 


Cheers 
Rob
